workitem/link/link_repository.go

```diff
@@ -97,6 +97,31 @@ func (r *GormWorkItemLinkRepository) ValidateCorrectSourceAndTargetType(ctx cont
 	return nil
 }
 
+// CheckParentExist returns error if there is an attempt to create more than 1 parent of a workitem.
+func (r *GormWorkItemLinkRepository) CheckParentExist(ctx context.Context, sourceID, targetID uint64, linkTypeID uuid.UUID) error {
```

Comment:
Maybe `ParentExists(...)` would be better. I am also fine with the current name with a slight change: `CheckParentExists(...)`.

I will try to elaborate my concern about the API.  In Go, there is no "exception handling". The commonly used pattern is to return two or more values where the last value will be an error value. Here is an example API: https://golang.org/pkg/strconv/#Atoi As you can see `Atoi` return the value and an error if any. In this case, if there is a DB error, that doesn't say anything about the existence of a parent, so this is an error scenario with the value as `nil`.

So, these are the possible return values:
```
nil, err
false, nil
true, nil
```
Now, on the calling side, if `err` value is not nil, return an error. Also if the function return `(true, nil)`, return an error as this is not expected.

This makes `ParentExists(...)` generic and it can be used in another controller where `(true, nil)` may not return an error.